### PR TITLE
topology: sof-icl-rt711-rt1308-rt715: refine dai index setting

### PR DESCRIPTION
**Pull request by RanderWang**
Refine topology according to the changes in machine driver.

Machine driver PR: https://github.com/thesofproject/linux/pull/1705

### CODE REVIEW

File: tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4 [hunk 1]
```diff
@@ -169,10 +171,10 @@ PCM_PLAYBACK_ADD(Headphone, 0, PIPELINE_PCM_1)
...
 PCM_PLAYBACK_ADD(SDW1-speakers, 2, PIPELINE_PCM_3)
 ifdef(`MONO', `', `PCM_PLAYBACK_ADD(SDW2-speakers, 3, PIPELINE_PCM_4)')
-PCM_CAPTURE_ADD(DMIC, 4, PIPELINE_PCM_5)
-PCM_PLAYBACK_ADD(HDMI1, 5, PIPELINE_PCM_6)
-PCM_PLAYBACK_ADD(HDMI2, 6, PIPELINE_PCM_7)
-PCM_PLAYBACK_ADD(HDMI3, 7, PIPELINE_PCM_8)
+PCM_CAPTURE_ADD(DMIC, DMIC_ID, PIPELINE_PCM_5)
+PCM_PLAYBACK_ADD(HDMI1, eval(DMIC_ID + 1), PIPELINE_PCM_6)
+PCM_PLAYBACK_ADD(HDMI2, eval(DMIC_ID + 2), PIPELINE_PCM_7)
+PCM_PLAYBACK_ADD(HDMI3, eval(DMIC_ID + 3), PIPELINE_PCM_8)
```
Comment (plbossart) [inline tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4]: what kernel PR changes this? We should merge only when both kernel and topology are ready

Comment (lgirdwood) [reply tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4]: @RanderWang please link this PR to your corresponding kernel PR.

**Comment by RanderWang** [comment]
Thanks, I update the PR for machine driver in PR description

**Comment by lgirdwood** [comment]
> Thanks, I update the PR for machine driver in PR description

Ok, can you link to here. i.e. paste web link for kernel PR in a comment here.

**Comment by lgirdwood** [comment]
@plbossart let us know if kernel PR is fine and we can then merge this one.

**Comment by zrombel** [comment]
Due to merge of PR#2303 this PR must be rebased or test results for KD tests with audio format 24b/32b ignored.

**Comment by jajanusz** [comment]
@RanderWang Please fix conflicts, or close PR if you don't need it anymore

**Comment by plbossart** [comment]
> @RanderWang Please fix conflicts, or close PR if you don't need it anymore

it's needed and tied to the machine driver changes mentioned above. We'll update both in the next week.

**Comment by RanderWang** [comment]
Update my PR, thanks!

**Pull request merged**